examples/sbol2/CreatingSBOL2Objects/Component.ipynb

```diff
@@ -0,0 +1,205 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# --- First Component: Promoter ---\n",
+    "# Step 1: Ceate a ComponentDefinition object for the promoter\n",
+    "promoter_component_definition = sbol2.ComponentDefinition('promoter_component_definition', component_type=sbol2.BIOPAX_DNA)\n",
+    "promoter_component_definition.addRole(sbol2.SO_PROMOTER)\n",
+    "doc.addComponentDefinition(promoter_component_definition)\n",
+    "\n",
+    "# Step 2: Create a Component object for the promoter\n",
+    "promotor_component = sbol2.Component('promoter_comoponent')\n",
+    "\n",
+    "# Step 3: Connect the Component object to the ComponentDefinition object through the definition property\n",
+    "promotor_component.definition = promoter_component_definition\n",
+    "\n",
+    "# Step 4: Add the Component to the the components list in the parent ComponentDefinition\n",
+    "genetic_construct_cd.components.add(promotor_component)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can also specify which elements of the gene-construct `ComponentDefinition` is to be included in the promoter. Let us say that the promoter spans the first 26 DNA nucleotides of the gene-construct."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "# Create a range objects to specify the location of the promotor on the genetic construct sequence\n",
+    "promotor_range = sbol2.Range('promoter_range', start=1, end=26)\n",
+    "promotor_component.sourceLocations.add(promotor_range)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Creating the CDS `Component`\n",
+    "We will now create a second Sub-Component for the gene-construct: A CDS. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "# --- Second Component: CDS ---\n",
+    "# Step 1: Ceate a ComponentDefinition object for the promoter\n",
+    "cds_component_definition = sbol2.ComponentDefinition('cds_component_definition', component_type=sbol2.BIOPAX_DNA)\n",
+    "cds_component_definition.addRole(sbol2.SO_CDS)\n",
+    "\n",
+    "doc.addComponentDefinition(cds_component_definition)\n",
+    "\n",
+    "# Step 2: Create a Component object for the promoter\n",
+    "cds_component = sbol2.Component('cds_component')\n",
+    "\n",
+    "# Step 3: Connect the Component object to the ComponentDefinition object through the definition property\n",
+    "cds_component.definition = cds_component_definition\n",
+    "\n",
+    "# Step 4: Add the Component to the the components list in the parent ComponentDefinition\n",
+    "genetic_construct_cd.components.add(cds_component)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we will add a `role` and a `roleIntegration` to the cds_component"
```

Comment:
In this part of the notebook, I would like to have an example of a `component` that has its `role` property set along with a `roleIntegration` of `OverrideRoles`. However, I do not know what example scenario to use to demonstrate this functionality. I was thinking perhaps I could use an activator `ComponentDefinition` that is actually being used as a repressor. However, I'm not sure if this example makes sense. In addition, there does not seem to be predefined pysbol constants of `Activator` and `Repressor` so I think I would have to find the relevant ontology and create my own URI. How would I go about doing this? Is this the path I should be going down? Any assistance would be appreciated.

Reply:
The `overrideRoles` value is very rarely used, and I'm not sure we have many good examples. I don't think that I've ever used it myself.

@cjmyers , do you have a good example here?